src/plugins/share/public/plugin.ts

```diff
@@ -86,7 +86,7 @@ export class SharePlugin implements Plugin<SharePluginSetup, SharePluginStart> {
     const { basePath } = http;
 
     this.url = new UrlService({
-      baseUrl: basePath.publicBaseUrl || basePath.serverBasePath,
+      baseUrl: basePath.get(),
```

Comment:
This is the fix, use `get()` so that baseUrl includes current space. 

Looks like In case we'll need to navigate between spaces using URL service or if we'll need to use URL service on the server with custom space - then some change will need to be done to URL service